File: scripts/map.js
// class Map
function Map(w, h)
{
  this.weight = w;
  this.height = h;
  this.map = [];
  
  this.Create = function()
  {
    for(var i = 0; i < this.height; i++)
    {
      this.map[i] = [];
      for(var j = 0; j < this.weight; j++)
      {
        this.map[i][j] = 0;
      }
    }
  }
  
  this.Open = function(nam)
  {
     //open from file
  }
  
  this.CheckPosition = function(pos)
  {
    return map[pos.x][pos.y] > 0;
  }
}

File: Map.cs
using LibFunc;

namespace snakegame
{
    class Map
    {
        int weight;
        int height;
        int[,] map;

        public Map(int w, int h)
        {
            weight = w;
            height = h;
            map = new int[h, w];
            Create();
        }

        public void Create()
        {
            for (int i = 0; i < height; i++)
            {
                for (int j = 0; j < weight; j++)
                {
                    map[i, j] = 0;
                }
            }
        }

        public void Open(string name)
        {

        }

        bool CheckPosition(Point pos)
        {
            return map[pos.X, pos.Y] > 0;
        }
    }
}
